src/calculator.py

```diff
@@ -168,9 +169,14 @@ def main():
             sess.commit()
             
         sess.close()
+        
+        if killhandler.killed:
+            logging.info("Termination signal received - exiting")
+            break
 
 
 if __name__ == "__main__":
+    killhandler = KillHandler()
```

Comment:
We need to move the instantiation of `killhandler` into the `main()` function where it's referenced. We should also remove this outermost `while` loop--uncaught exceptions should just crash the container (which is rebooted automatically) instead of getting swallowing like this.

Reply:
(And yeah I know the funky stuff in this outer loop wasn't added by you, but you're already making changes to the area so I thought I'd ask :-) )